rfcs/text/0000-multi-data-collection-dataset-processing.md

```diff
@@ -0,0 +1,133 @@
+
+For full background and a detailed proposal on how these libraries are represented in the metadata see the RFC [Representing sequencing library preparations in the HCA DCP metadata standard](https://github.com/HumanCellAtlas/dcp-community/blob/68f0c42fc9bd55297be8c241663721271c90b694/rfcs/text/0010-rfc-library-preparation.md).
+
+The *data group* submission event does not define bundles to be co-processed, it indicates that data to be co-processed is contained within the *data group* and that that group fulfills version-completeness requirements. Co-processing will be initiated by pipelines based on a well-defined set of rules that is data modality-specific.  The partition of a submission *data group* into co-processing units may require multiple parameters. For instance, with 10X V2 3’ scRNA-seq is grouped by library, however, a submission may contain non-10X data modalities that need to be processed separately.
+
+An analysis pipeline event handler component will be created that is responsible for partitioning a *data group* into co-processing units, dispatching the pipelines, and tracking completion. This is the responsibility of Analysis.  When analysis results are submitted to ingest, this creates a secondary analysis *data group*.
+
+The creation and update of *PROJECT_SUBMISSIONS* *data groups*, both primary and secondary, is the responsibility of the Ingest system.
+
+![Figure 2](../images/0000-multi-data-collection-data-processing-images/flow-of-information.png)
+Figure 2: Flow of information in the proposed data model
+
+#### Updates to *data groups*
+Updates to *PROJECT_SUBMISSIONS* must result in updating of the associated *data groups* to trigger reprocessing. When part or the entirety of a project submission is updated the associated data group must be identified and updated with the new bundle versions if bundles have been replaced and/or with additional new bundles if bundles have been added. This is the responsibility of Ingest.  Analysis must handle the update to the *data group* by initiating only the pipelines that have been affected by the update and ensure that the output bundles of this analysis are updates to existing bundles.
+
+*Data group* updates may result from creating new bundles to add additional types of data to a project, or updating bundles to change metadata or append data to data files (e.g. FASTQ top-up).
+
+#### Implementation feasibility
+
+Both Analysis and Ingest confirm that implementation of the data_groups bundle proposed here is feasible in principle. Analysis indicates that significant changes need to be made to the existing infrastructure to handle the new type of notification. The generation of secondary data_groups of type PROJECT_SUBMISSION will be generated by analysis submitting to submission and Ingest will identify the bundles included in the secondary bundle.
```

Comment:
"analysis submitting to submission" -> "analysis making a single submission to Ingest"

Reply:
Sounds like this needs clarification because my understanding was that analysis would do a submission per pipeline, and then ingest would track and identify the bundles to be included in the secondary analysis data group.

Reply:
Right. I'm not sure if it can work the way you describe since ingest would have no way to determine what goes in a group. Analysis would still need to signal this which means a new API mechanism. If you can't submit everything at once then we'd need to look at this in more detail, but that still means you need to track information anyway.

Reply:
Ingest is creating the project submission groups to begin with, so wouldn't ingest know what belongs in the group?

Reply:
We just calcified the text as the will be a project group for each level in the data processing pipeline.  However, that doesn't actually invalidate your point, as ingest could look it up.

The underlying issue is that there isn't a 1-to-1 mapping of analysis to primary data bundles, ingest doesn't know the mapping and doesn't know when all of the input bundles have been analyzed.  t somehow has to be told.  In the case of the primary data, the produce of the data tells ingest the bounds of the data.  Analysis will have to do the same.

Sorry ...